app/views/devise/two_factor_authentication_setup/confirm.html.slim

```diff
@@ -2,5 +2,5 @@
 
 h2 = t('devise.two_factor_authentication.enter_auth_code')
 = form_tag([resource_name, :two_factor_authentication], method: :put, role: 'form') do
-  = text_field_tag :code
+  = number_field_tag :code, input_html: { autofocus: true, type: :number }
```

Comment:
I believe the `input_html` is not necessary here. I think you can do:
```
= number_field_tag :code, autofocus: true
```
`type: number` would be redundant here since the Rails `number_field_tag` helper automatically adds that.